docs/topic_guides/choosing_comparators.ipynb

```diff
@@ -0,0 +1,1249 @@
+       "      <td>[S30105, R02063]</td>\n",
+       "      <td>[STFN, RXRT]</td>\n",
+       "      <td>[(STFN, ), (RXRT, RKRT)]</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "    string1    string2            soundex     metaphone  \\\n",
+       "0   Stephen    Stephen   [S30105, S30105]  [STFN, STFN]   \n",
+       "1   Stephen     Steven   [S30105, S30105]  [STFN, STFN]   \n",
+       "2   Stephen    Stephan   [S30105, S30105]  [STFN, STFN]   \n",
+       "3   Stephen      Steve    [S30105, S3010]   [STFN, STF]   \n",
+       "4   Stephen    Stehpen   [S30105, S30105]  [STFN, STPN]   \n",
+       "5   Stephen    tSephen   [S30105, t50105]  [STFN, TSFN]   \n",
+       "6   Stephen    Stephne  [S30105, S301050]  [STFN, STFN]   \n",
+       "7   Stephen     Stphen    [S30105, S3105]  [STFN, STFN]   \n",
+       "8   Stephen    Stepheb   [S30105, S30101]  [STFN, STFP]   \n",
+       "9   Stephen  Stephanie  [S30105, S301050]  [STFN, STFN]   \n",
+       "10  Stephen    Richard   [S30105, R02063]  [STFN, RXRT]   \n",
+       "\n",
+       "                  dmetaphone  \n",
+       "0       [(STFN, ), (STFN, )]  \n",
+       "1       [(STFN, ), (STFN, )]  \n",
+       "2       [(STFN, ), (STFN, )]  \n",
+       "3        [(STFN, ), (STF, )]  \n",
+       "4       [(STFN, ), (STPN, )]  \n",
+       "5       [(STFN, ), (TSFN, )]  \n",
+       "6       [(STFN, ), (STFN, )]  \n",
+       "7       [(STFN, ), (STFN, )]  \n",
+       "8       [(STFN, ), (STFP, )]  \n",
+       "9       [(STFN, ), (STFN, )]  \n",
+       "10  [(STFN, ), (RXRT, RKRT)]  "
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ch.phonetic_transform_df(data, \"string1\", \"string2\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Combining String scores and Phonetic matching\n",
+    "\n",
+    "Once you have considered all of the string comparators and phonetic transforms for a given column, you may decide that you would like to have multiple comparison levels including a combination of options.\n",
+    "\n",
+    "For this you can construct a custom comparison to catch all of the edge cases you want. For example, if you decide that the comparison for `first_name` in the model should consider:\n",
+    "\n",
+    "1. A `Dmetaphone` level for phonetic similarity\n",
+    "2. A `Levenshtein` level with distance of 2 for typos\n",
+    "3. A `Jaro-Winkler` level with similarity 0.8 for fuzzy matching\n",
+    "\n",
+    "The [name_comparison](../comparison_template_library.md#NameComparisonBase) function from the `Comparison Template Library` can be configured as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
```

Comment:
I think it's worth putting `first_name_comparison.as_dict()` as a separate code block and then put a line above to contextualise the SQL that is in the text block. Something like... "If we print this comparison as a dictionary we can see the underlying SQL."